src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -100,6 +100,18 @@ public void addRow(PrimaryKey key, Row row, long sstableRowId) throws IOExceptio
         }
     }
 
+    @Override
+    public void onSSTableWriterSwitched(Stopwatch stopwatch) throws IOException
+    {
+        if (maybeAbort())
+            return;
+
+        boolean emptySegment = currentBuilder == null || currentBuilder.isEmpty();
+        logger.debug("Flushing index {} with {}buffered data on sstable writer switched...", index.identifier().indexName, emptySegment ? "no " : "");
+        if (!emptySegment)
+            flushSegment();
```

Comment:
Note that we don't have to aggressively create a new builder here because `addTerm()`, `complete()`, and `abort()` will all handle the `currentBuilder == null` case appropriately.